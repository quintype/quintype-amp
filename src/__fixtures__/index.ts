import { config, publisherConfig, ampConfig, configOpts } from "./config.fixture";
import { textStory } from "./text-story.fixture";
import { storyWithManyJsEmbeds } from "./story-with-jsembeds.fixture";
import { relatedStories } from "./related-stories";
import { seo } from "./seo.fixture";
import { allElementsStory } from "./all-element-story.fixture";
import { liveBlogStory } from "./live-blog-story.fixture";

export {
  allElementsStory,
  config,
  publisherConfig,
  ampConfig,
  textStory,
  storyWithManyJsEmbeds,
  relatedStories,
  configOpts,
  seo,
  liveBlogStory
};
