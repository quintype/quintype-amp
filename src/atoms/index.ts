import Carousel from "./carousel";
import { Image } from "./image";
import Layout from "./layout";
import Facebook from "./facebook";
import Twitter from "./twitter";
import { Footer } from "./footer";
import { Section } from "./section";
import { Spacer } from "./spacer";
import { Author } from "./author";
import { DateTime } from "./date-time";
import { StoryElement } from "./story-elements/story-element";
import { StoryElements } from "./story-elements";

export {
  Carousel,
  Image,
  Layout,
  Facebook,
  Twitter,
  Footer,
  Section,
  Spacer,
  Author,
  DateTime,
  StoryElement,
  StoryElements
};
