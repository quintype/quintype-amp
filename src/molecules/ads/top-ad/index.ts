import { TopAd } from "./top-ad";
export { TopAd };
