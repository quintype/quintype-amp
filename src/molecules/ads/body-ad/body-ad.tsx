import React from "react";
import { withConfig } from "../../../context";
import { DfpAd, Spacer } from "../../../atoms";
import { getPropsForDfpAd } from "../shared/helpers";
import { CommonDfpAdTypes } from "../shared/types";
import { AdWrapper } from "../shared/components";
import get from "lodash.get";

export const BodyAdBase = ({ children, config, templateName, ...overridingProps }: CommonDfpAdTypes) => {
  const propsForBodyAd = getPropsForDfpAd({ overridingProps, config, adName: "body-ad" });
  const enabled = !!propsForBodyAd && get(config, ["opts", "featureConfig", "enableAds", templateName, "body"], true);
  return (
    enabled && (
      <AdWrapper>
        <Spacer token="s" />
        <DfpAd {...propsForBodyAd}>{children}</DfpAd>
        <Spacer token="s" />
      </AdWrapper>
    )
  );
};

export const BodyAd = withConfig(BodyAdBase);
