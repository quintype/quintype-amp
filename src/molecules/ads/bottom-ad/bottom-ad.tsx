import React from "react";
import { withConfig } from "../../../context";
import { DfpAd } from "../../../atoms";
import { getPropsForDfpAd } from "../shared/helpers";
import { CommonDfpAdTypes } from "../shared/types";
import { AdWrapper } from "../shared/components";

const BottomAdBase = ({ children, config, ...overridingProps }: CommonDfpAdTypes) => {
  const propsForTopAd = getPropsForDfpAd({ overridingProps, config, adName: "bottom-ad" });
  return (
    <AdWrapper>
      <DfpAd {...propsForTopAd}>{children}</DfpAd>
    </AdWrapper>
  );
};

export const BottomAd = withConfig(BottomAdBase);
