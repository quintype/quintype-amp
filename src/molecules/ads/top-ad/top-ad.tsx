import React from "react";
import { withConfig } from "../../../context";
import { DfpAd } from "../../../atoms";
import { getPropsForDfpAd } from "../shared/helpers";
import { CommonDfpAdTypes } from "../shared/types";
import { AdWrapper } from "../shared/components";

const TopAdBase = ({ children, config, ...overridingProps }: CommonDfpAdTypes) => {
  const propsForTopAd = getPropsForDfpAd({ overridingProps, config, adName: "top-ad" });
  return (
    propsForTopAd && (
      <AdWrapper>
        <DfpAd {...propsForTopAd}>{children}</DfpAd>
      </AdWrapper>
    )
  );
};

export const TopAd = withConfig(TopAdBase);
