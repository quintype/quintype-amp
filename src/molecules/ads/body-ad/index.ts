import { BodyAd } from "./body-ad";
export { BodyAd };
