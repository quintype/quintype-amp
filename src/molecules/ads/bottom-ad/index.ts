import { BottomAd } from "./bottom-ad";
export { BottomAd };
