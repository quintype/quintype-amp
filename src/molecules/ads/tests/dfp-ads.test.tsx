import React from "react";
import { mount } from "enzyme";
import cloneDeep from "lodash.clonedeep";
import { TopAd } from "../top-ad";
import { BodyAd } from "../body-ad";
import { BottomAd } from "../bottom-ad";
import { Layout } from "../../../atoms";
import { textStory, config } from "../../../__fixtures__";

const dummyConfig = cloneDeep(config);
dummyConfig.ampConfig.doubleclick = {
  "top-ad": {
    width: 300,
    height: 250,
    "unit-path": "/4444/lorem/ipsum/sticky"
  },
  "body-ad": {
    width: 400,
    height: 100,
    "unit-path": "/3333/lorem/ipsum/sticky"
  },
  "bottom-ad": {
    width: 750,
    height: 50,
    "unit-path": "/2222/lorem/ipsum/sticky"
  }
};

describe("DFP Ads", () => {
  it("should render TopAd, BodyAd, BottomAd correctly", () => {
    const wrapper = mount(
      <Layout story={textStory} config={dummyConfig}>
        <TopAd />
        <BodyAd />
        <BottomAd />
      </Layout>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("amp-ad").length).toBe(3);
    wrapper.find("amp-ad").forEach((node, index) => {
      switch (index) {
        case 0:
          expect(node.prop("width")).toBe(300);
          expect(node.prop("height")).toBe(250);
          expect(node.prop("data-slot")).toBe("/4444/lorem/ipsum/sticky");
          expect(node.prop("json")).toBe(
            `{"targeting":{"environment":["production"],"publisher-name":["vikatan"],"sections":["news"]}}`
          );
          break;
        case 1:
          expect(node.prop("width")).toBe(400);
          expect(node.prop("height")).toBe(100);
          expect(node.prop("data-slot")).toBe("/3333/lorem/ipsum/sticky");
          expect(node.prop("json")).toBe(
            `{"targeting":{"environment":["production"],"publisher-name":["vikatan"],"sections":["news"]}}`
          );
          break;
        case 2:
          expect(node.prop("width")).toBe(750);
          expect(node.prop("height")).toBe(50);
          expect(node.prop("data-slot")).toBe("/2222/lorem/ipsum/sticky");
          expect(node.prop("json")).toBe(
            `{"targeting":{"environment":["production"],"publisher-name":["vikatan"],"sections":["news"]}}`
          );
          break;
      }
    });
  });
  it("TopAd, BodyAd, BottomAd should apply overriding props", () => {
    const customPropsForAd = {
      width: "320",
      type: "highlyAnnoyingAd",
      "data-slot": "lorem/ipsum"
    };
    const wrapper = mount(
      <div>
        <Layout story={textStory} config={dummyConfig}>
          <TopAd {...customPropsForAd} />
          <BodyAd {...customPropsForAd} />
          <BottomAd {...customPropsForAd} />
        </Layout>
      </div>
    );
    wrapper.find("amp-ad").forEach((node) => {
      expect(node.prop("data-slot")).toBe("lorem/ipsum");
      expect(node.prop("type")).toBe("highlyAnnoyingAd");
      expect(node.prop("width")).toBe("320");
    });
  });
});
