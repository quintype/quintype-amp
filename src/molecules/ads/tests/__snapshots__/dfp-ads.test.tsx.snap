// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopAd snapshot test bodyAd 1`] = `
<BodyAdBase
  config={Object {}}
/>
`;

exports[`TopAd snapshot test bottomAd 1`] = `
<BottomAdBase
  config={Object {}}
/>
`;

exports[`TopAd snapshot test topAd 1`] = `
<TopAdBase
  config={Object {}}
/>
`;
